client/packages/requisitions/src/ResponseRequisition/DetailView/ResponseLineEdit/ResponseLineEdit.tsx

```diff
@@ -74,187 +89,333 @@ export const ResponseLineEdit = ({
   return (
     <Box>
       <Box display="flex" justifyContent="space-between">
-        <Box paddingLeft={4} paddingRight={7}>
-          {/* Left column content */}
-          {!isProgram ? (
-            <InputWithLabelRow
-              Input={
-                <NumericTextInput
-                  width={INPUT_WIDTH}
-                  value={draft?.availableStockOnHand}
-                  onChange={value => update({ availableStockOnHand: value })}
-                  onBlur={save}
-                  disabled={!!hasLinkedRequisition}
-                  autoFocus
-                />
+        {isNew ? (
+          <Box width="100%">
+            <StockItemSearchInput
+              onChange={(newItem: ItemRowFragment | null) => {
+                if (newItem) {
+                  insert({
+                    id: FnUtils.generateUUID(),
+                    requisitionId: requisitionId,
+                    itemId: newItem.id,
+                  });
+                  navigate(buildItemEditRoute(requisitionNumber, newItem.id));
+                }
+              }}
+              openOnFocus={true}
+              extraFilter={item =>
+                !lines.some(line => line.item.id === item.id)
               }
-              labelWidth={LABEL_WIDTH}
-              label={t('label.stock-on-hand')}
-              sx={{ marginBottom: 1 }}
             />
-          ) : (
-            <InputWithLabelRow
-              Input={
-                <NumericTextInput
-                  width={INPUT_WIDTH}
-                  value={draft?.initialStockOnHandUnits}
-                  onChange={value => update({ initialStockOnHandUnits: value })}
-                  onBlur={save}
-                  disabled={!!hasLinkedRequisition}
-                  autoFocus
+          </Box>
+        ) : (
+          <>
+            <Box paddingLeft={4} paddingRight={7}>
+              {/* Left column content */}
```

Comment:
same here, is it just shuffling? sorry not familiar with what should/shouldn't be there 😁